### ISSUE TITLE
python: conditional markdown injection for comments

### ISSUE THREAD

**Issue by cgahr** [helix-editor/helix#9836]
In python, you can create jupyter notebooks in the `# %%` format. They looks as follows:
```python
# %% [markdown]
# # Formulas
# Let $m = \max(p, r, q)$. Then, we define the matrix $\Psi_k \in \mathbb{R}^{(N - m) \times n_0}$ as
# $$
# \Psi_k = \begin{pmatrix} | & | & & | \\ \psi_k^{m+1} & \psi_k^{m+2} & \ldots & # \psi_k^N \\ | & | & & | \end{pmatrix}^T.
# $$

# %%
print("And here is some python code!")
```
In this format, cells are delimited by `# %%`. `# %% [markdown]` cells contain comments, whose content is interpreted as `markdown` by removing the leading `#`.

### What I would like to have

If a comment starts with `# %% [markdown]`, parse comments until the next `# %%` as markdown. Is this even possible? Or must the python-tree-sitter change to accomodate this?

The closest I got is replacing `comment` with `markdown` in `runtime/queries/python/injections.scm`:
```
((comment) @injection.content
 (#set! injection.language "markdown"))
```
However, then every comment is parsed as markdown.

Furthermore, the latex content in the first cell, is not highlighted properly; if I remove the leading `#` of the first cell and set the language to `markdown`, the latex code is highlighted properly.

If I inject `markdown.inline` instead, formulas in a single line, are highlighted properly, i.e. `# $\theta$` works, however, formulas split between multiples lines are not highlighted properly.

## Basically, I want 2 things:
- conditional syntax highlighting: if a comment start with `# %% [markdown]`, parse the comment until the next `# %%` as markdown
- inject and highlight `latex` into `markdown` into `python` correctly

**Comment by kirawi** [comment]
I don't think it's possible to do that with comments without the grammar recognizing them.

**Comment by cgahr** [comment]
I was playing around with this and (maybe) found a neat solution: I created a new grammar. This grammar parses each cell, either as `code_cell` or `markdown_cell`. Depending on the cell, I can then inject the proper language highlighting. This should give me the desired effect (I still have to tey this).

However, I foresee a very big issue: since the underlying language is not python anymore, indent-queries and movement (e.g. go to function) won't work.

Is there a way around this? I.e. do injections not only add syntax highlighting but also indent-queries etc.?
